server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/aggregators/numerictype/StarTreeNumericTypeConverters.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube.startree.aggregators.numerictype;
+
+import org.apache.lucene.sandbox.document.HalfFloatPoint;
+import org.apache.lucene.util.NumericUtils;
+
+public class StarTreeNumericTypeConverters {
```

Comment:
We need not tie to star tree, can we check if we can have a generic util which gets utilized in both star tree and in the places where this conversion happens

Reply:
As discussed offline, keeping this as separate for now to reduce the blast radius.

Reply:
We are keeping it separate yes but can we use generic util instead of tying these methods to star tree and reuse across star tree and existing mappers ?